### ISSUE TITLE
PySerial only working from interactive python, not from script in windows

### ISSUE THREAD

**Issue by leonhardpetzel** [pyserial/pyserial#735]
Hello,

I found a really strange issue on Windows 10 Pro using python 3.12.1 with PySerial 3.5.

I have an arduino board with MEGA2560 connected via USB and the firmware on the board will reply to a "*IDN?\r" command with some version string.

If I send the commands in an interactive python promt, all works fine:
![Python Bug 0](https://github.com/pyserial/pyserial/assets/100354556/b88ba644-83f6-47dd-ba25-5a58583c6979)

Also from the debug console (in VS Code) all works fine:
![Python Bug 1](https://github.com/pyserial/pyserial/assets/100354556/8bb04e75-4098-42e3-9f60-d6ea43b449e5)

However, when I run the commands from a script, it seems like nothing is sent to the board:
![Python Bug 2](https://github.com/pyserial/pyserial/assets/100354556/72fc268d-093b-464b-9f61-795ea1627e96)

I tried to put time.sleep() everywhere, but it did not change anything. It really seems like the difference is, if the call comes from an interactive session, or not.

I am not sure, if the issue originates from PySerial, from python or from Windows.
Does somebody here have an idea, where this could come from?

Thanks and BR

**Comment by paratim** [comment]
Yes, I'm seeing an issue similar to yours. In my case a sleep(5) after the serial port creation seemed to be a workaround.

**Comment by leonhardpetzel** [comment]
> Yes, I'm seeing an issue similar to yours. In my case a sleep(5) after the serial port creation seemed to be a workaround.

Thanks for the information! A 2 second sleep works for me. Amazing! I only tried 100ms sleep, which I thought was super long, but apparently not enough.

I also found that writing to the .timeout member is a very bad idea. As soon as I try to change the timeout, the next send transaction just sends garbage.

**Comment by tay10r** [comment]
I'm glad I found this issue, it's happening to me as well. In my case (if it makes a difference), the device is an Arduino Uno R3.

**Comment by kurtmckee** [comment]
@zsquareplusc @andrewleech Please close this issue. It appears to be an issue with the device auto-rebooting, not an issue with pyserial.

Both here and elsewhere, [users report that the behavior of the Arduino Uno R3 is to auto-reset when the serial port is opened](https://forum.arduino.cc/t/prevent-usb-serial-connection-from-restarting-mega/1159077), which explains why a REPL or debug session can incidentally give the board enough time to reboot.
